src/app/pages/update-profile-request/components/update-action.tsx

```diff
@@ -16,20 +15,18 @@ import { gaiaUrl } from '@shared/constants';
 
 import { useAnalytics } from '@app/common/hooks/analytics/use-analytics';
 import { useCurrentStacksAccount } from '@app/store/accounts/blockchain/stacks/stacks-account.hooks';
-import { useStacksWallet } from '@app/store/accounts/blockchain/stacks/stacks-keychain';
 import { useProfileUpdateRequestSearchParams } from '@app/store/profiles/requests.hooks';
 
 import { UpdateActionLayout } from './update-action.layout';
 
 function useUpdateProfileSoftwareWallet() {
   const account = useCurrentStacksAccount();
-  const wallet = useStacksWallet();
 
   return useCallback(
     async (publicProfile: PublicPersonProfile) => {
       const fetchFn = createFetchFn();
 
```

Comment:
We seem to have `WalletType` and `account.type` that get used interchangeably. Maybe all `wallet`s can be `account`s ?

Reply:
Gist here is that the wallet can be of a certain mode unrelated to accounts themselves, but we also need a way to use discriminating unions to narrow the account type to either Ledger or Software.

Any suggestions how this could be more clear?